prow/github/types.go

```diff
@@ -538,6 +538,7 @@ type GenericCommentEvent struct {
 	Number       int
 	Repo         Repo
 	User         User
+	ReviewState  string
```

Comment:
I don't think we should add this field to the `GenericCommentEvent` type because it only makes sense for `pull_request_review` and `pull_request_review_comment` events.

Instead you should add an event handler for `pull_request_review` events to the `lgtm` plugin and move the common handling logic into a common function that is used by the generic comment handler and the review event handler. For an example, see the [`approve` plugin](https://github.com/kubernetes/test-infra/blob/master/prow/plugins/approve/approve.go). It handles `pull_request` and generic comment events, but uses a common handler for much of the logic. In this case the difference in the logic for different event types shouldn't be as drastic as the `approve` plugin, you will just be using a different indicator of whether the user wants to add or remove `lgtm`.

Reply:
Is the main advantage here to be logically consistent? If so, I'm happy to change the comment (also, the comments @joejulian refers to are stale anyway since much (all?) of the plugins are using guthub API v4 (not v3).

Reply:
@joejulian Generic comment events are synthetic events created for any type of text comment on Github. We use them to handle commands because it provides a general event that is created for any of the event types that involve user text so plugins that want to react to commands can handle a single event type instead of 5 different event types.
`pull_request_review` events should definitely be `GenericCommentEvent`s because they contain user text and could include commands. 

@guineveresaenger Yes, the main advantage is logical consistency. We shouldn't shove the review state information into `GenericCommentEvent`s because it isn't a property that most comments have. We really are trying to react to a review, not a comment so it makes more sense to handle a review event than to add review event info to `GenericCommentEvent`. That would be a step towards making it a catch all event type that represents more than just a generalized user comment.
Most of our plugins are actually still using Github's v3 API. The v4 API is not yet full featured and doesn't always provide an advantage over v3. We receive the same webhooks using either API though.